tests/src/Interop/ArrayMarshalling/SafeArray/SafeArrayNative.cpp

```diff
@@ -0,0 +1,651 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <xplatform.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include "platformdefines.h"
+
+//--------------------------------
+//	Start of SafeArray calls
+//--------------------------------
+
+extern "C" DLL_EXPORT BOOL STDMETHODCALLTYPE SafeArray_In(SAFEARRAY* psa)
```

Comment:
These tests are a bit much. Instead, I would look at the COM tests for arrays and see the basic validation that is needed.

Reply:
I'd actually prefer to keep the validation of all of the different components of the safe array since they're not something we have tests for anywhere else and they add a lot of complexity on top of traditional arrays.

Reply:
I am fine with the validation - although in this case, you are testing windows not the CLR, but the validation is fine. My criticism is that it is in every case. That is definitely not needed and should be limited to a single case.